src/programs/cp/cp.c

```diff
@@ -2,15 +2,16 @@
 
 #include <stdio.h>
 #include <stdlib.h>
-#include <unistd.h>
+#include <string.h>
 #include <sys/errno.h>
+#include <unistd.h>
 
 int main(int argc, char *argv[])
 {
-    const char *cp_path = "/bin/cp";
-    const char *clone_arg = "-c";
-    const char **new_argv = malloc(sizeof(char *) * (argc+2));
-    if (new_argv) {
+    char *cp_path = strdup("/bin/cp");
+    char *clone_arg = strdup("-c");
+    char **new_argv = malloc(sizeof(char *) * (argc+2));
+    if (cp_path && clone_arg && new_argv) {
```

Comment:
Putting all into one condition will not work well with the `perror()` below. The `errno` could have been overwritten by the next library function.

Reply:
```suggestion
    if (cp_path && clone_arg && new_argv) {
```